### PR TITLE
Add support for new channel type "DBS"

### LINKED ISSUE #801: Support new channel type "DBS"
#### Describe the problem
In Version 0.23.0 of MNE, support for the channel type DBS for Deep Brain Stimulation recordings was added (https://mne.tools/stable/whats_new.html#version-0-23-0), so it would likely be desirable to add support for writing and reading data with MNE-BIDS as well.

#### Describe your solution
I implemented the support in a branch and made a pull request with the number #800 

In the sidecar _ieeg.json file, DBS channels are not counted separately but counted together with SEEG Channels in "SEEGChannelCount", as specified in BIDS: "SEEGChannelCount": "Number of iEEG depth channels included in the recording..." (https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/04-intracranial-electroencephalography.html#sidecar-json-_ieegjson)

### PR DESCRIPTION
**Pull request by richardkoehler**
PR Description
--------------
closes #801 
In Version 0.23.0 of MNE, support for the channel type DBS for Deep Brain Stimulation recordings was added (https://mne.tools/stable/whats_new.html#version-0-23-0), so it would likely be desirable to add support for writing and reading data with MNE-BIDS as well.
In the sidecar _ieeg.json file, DBS channels are not counted separately but counted together with SEEG Channels in "SEEGChannelCount", as specified in BIDS: "SEEGChannelCount": "Number of iEEG depth channels included in the recording..." (https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/04-intracranial-electroencephalography.html#sidecar-json-_ieegjson)

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/tests/test_utils.py [hunk 1]
```diff
@@ -49,12 +49,12 @@ def test_handle_datatype():
...
     sampling_rate = 100
     data = random((n_channels, sampling_rate))
-    # datatype is given, check once for each datatype
-    channel_types = ['grad', 'eeg', 'ecog', 'seeg']
-    datatypes = ['meg', 'eeg', 'ieeg', 'ieeg']
-    for ch_type, datatype in zip(channel_types, datatypes):
+    channel_types = ['grad', 'eeg', 'ecog', 'seeg', 'dbs']
+    expected_modalities = ['meg', 'eeg', 'ieeg', 'ieeg', 'ieeg']
+    # do it once for each type ... and once for "no type"
+    for chtype, datatype in zip(channel_types, expected_modalities):
```
Comment (agramfort) [inline mne_bids/tests/test_utils.py:55]:
```suggestion
    for ch_type, datatype in zip(channel_types, expected_modalities):
```
can you avoid such cosmetic changes? ch_type comes from mne-python variable naming convention.

thx

**Review by agramfort — COMMENTED** [review]
can you also add a what's new entry? 🙏

File: doc/whats_new.rst [hunk 2]
```diff
@@ -42,6 +42,7 @@ Enhancements
 - :class:`mne_bids.BIDSPath` now has property getter and setter methods for all BIDS entities, i.e., you can now do things like ``bids_path.subject = 'foo'`` and don't have to resort to ``bids_path.update()``. This also ensures you'll get proper completion suggestions from your favorite Python IDE, by `Richard Höchenberger`_ (:gh:`786`)
 - :func:`mne_bids.write_raw_bids` now stores information about continuous head localization measurements (e.g., Elekta/Neuromag cHPI) in the MEG sidecar file, by `Richard Höchenberger`_ (:gh:`794`)
 - :func:`mne_bids.write_raw_bids` gained a new parameter `empty_room` that allows to specify an associated empty-room recording when writing an MEG data file. This information will be stored in the ``AssociatedEmptyRoom`` field of the MEG JSON sidecar file, by `Richard Höchenberger`_ (:gh:`795`)
+- Added support for the new channel type `'dbs'`, which was introduced in MNE Version 0.23.0, by `Richard Köhler`_ (:gh:`800`)
```
Comment (hoechenberger) [inline doc/whats_new.rst:45]:
```suggestion
- Added support for the new channel type `'dbs'` (Deep Brain Stimulation), which was introduced in MNE-Python 0.23, by `Richard Köhler`_ (:gh:`800`)
```

File: mne_bids/tests/test_utils.py (hunk 1 above)
Comment (richardkoehler) [reply mne_bids/tests/test_utils.py]: Sorry, I think this change actually came from a rebase gone wrong, definitely wasn't intentional!

Comment (richardkoehler) [reply mne_bids/tests/test_utils.py]: Also added a whats_new entry now

**Review by hoechenberger — COMMENTED** [review]
Looks good to me! Do you think you could add an example too? Do we have an actual DBS dataset somewhere that we could use?

**Comment by richardkoehler** [comment]
> 
> 
> Looks good to me! Do you think you could add an example too? Do we have an actual DBS dataset somewhere that we could use?

Sure, I can definitely add an example. I was planning on making a short DBS tutorial for MNE-Python anyway, just haven't had the time in the last couple of weeks. I will select some data from a few recordings to provide an example DBS dataset.

**Review by hoechenberger — APPROVED** [review]
Great!

It's okay for me to merge this without an example for now, we can add it later. 👌

**Review by agramfort — APPROVED** [review]
thx @richardkoehler !

**Review by sappelhoff — APPROVED** [review]
thanks!

**Comment by hoechenberger** [comment]
Thank you!

**Pull request merged**